utils/utils.c

```diff
@@ -192,18 +192,23 @@ static bool is_uftrace_directory(const char *path)
 	fd = open(info_path, O_RDONLY);
 	free(info_path);
 
-	if (fd == -1)
-		return false;
-
-	if (read(fd, sig, UFTRACE_MAGIC_LEN) != UFTRACE_MAGIC_LEN) {
-		/*
-		 * partial read() will return false anyway
-		 * since memcmp() below cannot success.
-		 */
+	if (fd != -1) {
+		if (read(fd, sig, UFTRACE_MAGIC_LEN) != UFTRACE_MAGIC_LEN) {
+			/*
+			 * partial read() will return false anyway
+			 * since memcmp() below cannot success.
+			 */
+		}
+		close(fd);
+		return !memcmp(sig, UFTRACE_MAGIC_STR, UFTRACE_MAGIC_LEN);
 	}
 
-	close(fd);
-	return !memcmp(sig, UFTRACE_MAGIC_STR, UFTRACE_MAGIC_LEN);
+	/* if "info" file is missing, also check that there is "uftrace.opts" */
+	xasprintf(&info_path, "%s/uftrace.opts", path);
+	if (!access(info_path, F_OK))
+		return true;
```

Comment:
So what's the point of checking this?  I think it cannot work if info file is missing.

Reply:
I've seen many cases when running it on embedded devices. As of now, uftrace can be broken many times and it generates broken data, which doesn't have `info` file. If there is `uftrace.data` without `info` file. It will show some kind of unclear message as follows:
```
$ uftrace record t-abc
WARN: creating directory failed: File exists
```
Even if the user tries to record again, it just shows the same messages.
```
$ uftrace record t-abc
WARN: creating directory failed: File exists

$ uftrace record t-abc
WARN: creating directory failed: File exists

$ uftrace record t-abc
WARN: creating directory failed: File exists
```
It will make novice users got stuck and cannot do anything after that. So we better make another way to identify if it's uftrace directory or not. So I think `uftrace.opts` can be a good candidate because it's created right after the directory is created.